alacritty/src/daemon.rs

```diff
@@ -43,18 +49,44 @@ where
         .map(|_| ())
 }
 
+// get working directory of controlling process
+#[cfg(not(windows))]
+pub fn foreground_process_path() -> Result<PathBuf, Box<dyn Error>> {
+    let mut pid = unsafe { libc::tcgetpgrp(tty::master_fd()) };
+    if pid < 0 {
+        pid = tty::child_pid();
+    }
+
+    #[cfg(not(any(target_os = "macos", target_os = "freebsd")))]
+    let link_path = format!("/proc/{}/cwd", pid);
+    #[cfg(target_os = "freebsd")]
+    let link_path = format!("/compat/linux/proc/{}/cwd", pid);
+
+    #[cfg(not(target_os = "macos"))]
+    let cwd = fs::read_link(link_path)?;
+
+    // this might not be a io::Result, so a Boxed dyn Error is needed
+    #[cfg(target_os = "macos")]
+    let cwd = macos::proc::cwd(pid)?;
+
+    Ok(cwd)
+}
+
 #[cfg(not(windows))]
 fn spawn_daemon<I, S>(program: &str, args: I) -> io::Result<()>
 where
     I: IntoIterator<Item = S> + Copy,
     S: AsRef<OsStr>,
 {
+    let mut command = Command::new(program);
+    let mut command_builder = match foreground_process_path() {
+        Ok(cwd) => command.args(args).current_dir(cwd),
+        _ => command.args(args),
+    };
```

Comment:
`command.args(args)` is the same in both paths here? Can't you just move that out and use `if let` instead?

Reply:
Refactored.